spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/SftpSourceSessionFactoryConfiguration.java

```diff
@@ -37,17 +45,35 @@
 	@Bean
 	@ConditionalOnMissingBean
 	public SessionFactory<LsEntry> sftpSessionFactory(SftpSourceProperties properties, BeanFactory beanFactory) {
+		return buildFactory(beanFactory, properties.getFactory());
+	}
+
+	@Bean
+	@ConditionalOnProperty("sftp.multi-source")
```

Comment:
I wonder if we can somehow live without this property, because it looks like we can derive the boolean flag from the existing `directories` or `factories`.

There is a recent change in the Spring Framework that we can return `null` for the `@Bean` and seems for me this can work for the `@ConditionalOnBean` pretty nice.